lldb/source/Expression/DWARFExpression.cpp

```diff
@@ -2445,8 +2449,8 @@ bool DWARFExpression::Evaluate(
               return false;
             }
           }
-          op_piece_offset += piece_byte_size;
         }
+        op_piece_offset += piece_byte_size;
```

Comment:
This is the fix, moving the offset update outside of the `else` arm of `if (stack.empty()) {`.

You can see the "then" arm above (handling undef pieces) because a comment was added to it.